### DISCUSSION TITLE
Page Loaded event is triggered before DataContextChanged on non-Windows platforms

### DISCUSSION THREAD

**Discussion by LITTOMA** [unoplatform/uno#13701] [Q&A]
### Current behavior

``` csharp
public MyPage()
{
    this.InitializeComponent();
    this.DataContextChanged += (s, e) => Trace.WriteLine(">>>>> DataContextChanged");
    this.Loaded += (s, e) => Trace.WriteLine(">>>>> Loaded");
}
```

Output:
```
[0:] >>>>> ANDROID
[0:] >>>>> Loaded
[0:] >>>>> DataContextChanged
```

### Expected behavior

Output:
```
[0:] >>>>> ANDROID
[0:] >>>>> DataContextChanged
[0:] >>>>> Loaded
```

This is the sequence on Windows:
```
>>>>> WINDOWS
>>>>> DataContextChanged
>>>>> DataContextChanged
>>>>> DataContextChanged
>>>>> Loaded
>>>>> DataContextChanged
```

**Comment by jeromelaban** [comment]
Thanks for the report. 

This is a known issue, caused by https://github.com/unoplatform/uno/issues/3519. 

Until it's getting fixed, could you tell more about the scenario you're hitting this issue with? There's likely a workaround that can be used.

**Reply by LITTOMA** [reply]
I'm using `Microsoft.Xaml.Interactivity` and trying to execute a command on page loaded for loading some data automatically.

Currently my workaround is binding the command to `DataContextChanged` instead of `Loaded`. 
``` xaml
<i:Interaction.Behaviors>
    <ic:EventTriggerBehavior EventName="DataContextChanged">
        <ic:InvokeCommandAction Command="{Binding LoadDataCommand}" />
    </ic:EventTriggerBehavior>
</i:Interaction.Behaviors>
```
It can achieve my goal but if there's any better idea please let me know. 
Thank you.

**Reply by jeromelaban** [reply]
Indeed, if your actual dependency is on `DataContext` it's best to register on it as it can change. That being said, why do you need  to invoke a command on the same `DataContext` instance that was propagated?
